### PR TITLE
Refactor into a multi-module project

### PR DESCRIPTION
**Pull request by gdusbabek**
**No code was harmed while making this PR**

Create a parent pom and move all code into a submodule (gossip-base). Subsequent PRs can create new summodules, move more code, etc.

This is for GOSSIP-78.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,6 +41,26 @@
...
 	
+	<parent>
+		<groupId>org.apache</groupId>
+		<artifactId>apache</artifactId>
+		<version>RELEASE</version>
+	</parent>
+  
+  <name>Gossip Parent</name>
+  <groupId>org.apache.gossip</groupId>
+	<artifactId>gossip</artifactId>
```
Comment (edwardcapriolo) [inline pom.xml]: lets call the artifact gossip-parent

File: pom.xml [hunk 2]
```diff
@@ -77,9 +83,9 @@
 	
 	<dependencies>
```
Comment (edwardcapriolo) [inline pom.xml]: Should we move the dependencies into the module?

**Review by edwardcapriolo — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (gdusbabek) [reply pom.xml]: done in ebd8dddc3cd47cbde7189623a56a815608d99dd5

File: pom.xml (hunk 2 above)
Comment (gdusbabek) [reply pom.xml]: I'm on the fence. In other projects that have had global dependencies, I've left them in the parent. Most of these seem to fall into that category.

Comment (edwardcapriolo) [reply pom.xml]: In the end run there probably will be a few that are global and some that are specific. I do believe that the process should start with nothing in the global and then as things are shared by more than two modules they are pulled into the global

Comment (gdusbabek) [reply pom.xml]: No problem. Done.

**Pull request merged**